sdk/formrecognizer/ai-form-recognizer/review/ai-form-recognizer.api.md

```diff
@@ -0,0 +1,1299 @@
+export interface DocumentObjectField<Properties = {
+    [k: string]: DocumentField | undefined;
+}> extends DocumentFieldCommon {
+    // (undocumented)
+    kind: "object";
+    properties: Properties;
+}
+
+// @public
+export interface DocumentPage {
+    angle: number;
+    height: number;
+    lines: DocumentLine[];
+    pageNumber: number;
+    selectionMarks?: DocumentSelectionMark[];
+    spans: DocumentSpan[];
+    unit: LengthUnit;
+    width: number;
+    words: DocumentWord[];
+}
+
+// @public
+export interface DocumentPhoneNumberField extends DocumentFieldCommon {
+    // (undocumented)
+    kind: "phoneNumber";
+    value?: string;
+}
+
+// @public
+export interface DocumentSelectionMark {
+    boundingBox?: number[];
+    confidence: number;
+    span: DocumentSpan;
+    state: SelectionMarkState;
+}
+
+// @public
+export interface DocumentSelectionMarkField extends DocumentFieldCommon {
+    // (undocumented)
+    kind: "selectionMark";
+    value?: string;
+}
+
+// @public
+export interface DocumentSignatureField extends DocumentFieldCommon {
+    // (undocumented)
+    kind: "signature";
+}
+
+// @public
+export type DocumentSignatureType = string;
+
+// @public
+export interface DocumentSpan {
+    length: number;
+    offset: number;
+}
+
+// @public
+export interface DocumentStringField extends DocumentValueField<string> {
+    // (undocumented)
+    kind: "string";
+}
+
+// @public
+export interface DocumentStyle {
+    confidence: number;
+    isHandwritten?: boolean;
+    spans: DocumentSpan[];
+}
+
+// @public
+export interface DocumentTable {
+    boundingRegions?: BoundingRegion[];
+    cells: DocumentTableCell[];
+    columnCount: number;
+    rowCount: number;
+    spans: DocumentSpan[];
+}
+
+// @public
+export interface DocumentTableCell {
+    boundingRegions?: BoundingRegion[];
+    columnIndex: number;
+    columnSpan?: number;
+    content: string;
+    kind?: DocumentTableCellKind;
+    rowIndex: number;
+    rowSpan?: number;
+    spans: DocumentSpan[];
+}
+
+// @public
+export type DocumentTableCellKind = string;
+
+// @public
+export interface DocumentTimeField extends DocumentFieldCommon {
+    // (undocumented)
+    kind: "time";
+    value?: string;
+}
+
+// @public
+export interface DocumentValueField<T> extends DocumentFieldCommon {
+    value?: T;
+}
+
+// @public
+export interface DocumentWord {
+    boundingBox?: number[];
+    confidence: number;
+    content: string;
+    span: DocumentSpan;
+}
+
+// @public
+export type FieldSchema = StringLikeFieldSchema | NumberFieldSchema | DateFieldSchema | ArrayFieldSchema | ObjectFieldSchema | StructuredStringFieldSchema;
+
+// @public
+export type FormRecognizerApiVersion = "2021-09-30-preview";
+
+// @public
+export const FormRecognizerApiVersion: {
+    readonly Latest: "2021-09-30-preview";
+    readonly Preview: "2021-09-30-preview";
+};
+
+// @public
+export type FormRecognizerRequestBody = NodeJS.ReadableStream | Blob | ArrayBuffer | ArrayBufferView;
```

Comment:
Did you consider supporting https://streams.spec.whatwg.org/#rs-model instead of NodeJS's? [NodeJS will soon support it as well](https://nodejs.org/api/webstreams.html)

Reply:
I think it will be a while before we move wholesale away from nodejs readable streams. Too many of the APIs support it today to toss. Things like fs, http, compression APIs, and most 3rd party APIs that support transformations on a stream. Would not be opposed to supporting both, though last I checked there were issues with whatwg streams being used for request bodies in browsers.

Reply:
That is fair. My reasoning for the suggestion was basically that this package wants to be bleeding age in typescript features used given the major version bump freedom so it may as well consider bleeding edge input types. Supporting both types is good too and of course it is not a blocker for this PR.

Reply:
This is a manual redeclaration of `Exclude<RequestBodyType, string>`. String inputs are treated as URLs, so I just wanted to declare this as the alternative for a streamable body. Essentially this is what core-client declares it supports, minus string.

Reply:
We'll cross this bridge when we get there. I'd like to see core support for whatwg streams before adding them to FR so that we don't have to read them into a buffer in the convenience layer.

Reply:
Sounds good. Supporting whatwg streams in core will allow customers to use stream data in the browser too so I believe we should start thinking about doing it even though no one asked for it yet. Seems to be a net positive change imo.

Reply:
Agreed with phasing this in after we get it into core, which feels more and more like something we should tackle